### PR TITLE
Edit user

### PR DESCRIPTION
**Pull request by ancaolteanu**

### CODE REVIEW

File: logs/errors.log [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+  "lastName": "cii",
+  "email": "oanaa@gmail.com",
+  "phone": "0987604621"
+}
+[2018-05-31T17:04:42+03:00] PUT "/users/0" 400 OK - {
+  "firstName": "cipo",
+  "lastName": "cii",
+  "email": "oanaa@gmail.com",
+  "phone": "09876d04621"
+}
```
Comment (h-mihail) [inline logs/errors.log]: This should be added to ```.gitignore```

File: src/actions/editUserActions.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                    let errMessage = reqValidData(req.body, 1,users);
+                    res.json({ "err": errMessage }).status(400)
+                }
+            }
+
+        })
+        .catch(error => {
+            res.status(500);
+            res.json({
+                serverErrorMessage: "AICI 2! the error was logged and we’ll be checking it shortly"
```
Comment (h-mihail) [inline src/actions/editUserActions.js]: Did you forget to delete ```AICI``` and ```AICI 2!``` ?

**Review by DemetriadS — DISMISSED** [review]

File: src/actions/editUserActions.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+//metoda folosita in reqValidData pentru a verifica daca mailul deja exista
+let checkMail = (newMail, allMails) => {
+    for (let i = 0; i < allMails.length; i++) {
+        if (newMail === allMails[i].email) return 1;
+    }
+    return 0;
+};
+
+//metoda verifica daca elementele introduse de user sunt corecte si corespund campului corespunzator
+let reqValidData = (str, putMethod,users) => {
```
Comment (danutDiac) [inline src/actions/editUserActions.js]: These does something very similar with **dataValid** function from **createUserActions.js**. Please refactor before merge or create a new task in trello for this refactor

File: src/actions/editUserActions.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    }
+    else if (putMethod) {
+        ok += "Nu ati introdus niciun numar de telefon \n"
+    }
+
+    return (ok === "") ? 1 : ok;
+};
+
+
+//returneaza linia unde se afla id-ul cautat
```
Comment (danutDiac) [inline src/actions/editUserActions.js]: Comments usually indicate that the code itself is not self-descriptive. When you have the tendency to write comments thing about how would you name your function (or rewrite the code) to "say" what the comment says.
Ex: **getIdIndex** or **getIdIndexInArray**

File: src/actions/editUserActions.js (hunk 3 above)
Comment (Silv94) [reply src/actions/editUserActions.js]: Nu putem folosi functia dataValid din createUserActions.js pentru ca noi am implementat promise-uri si astfel apar discrepante in cod

File: src/actions/editUserActions.js (hunk 4 above)
Comment (ancaolteanu) [reply src/actions/editUserActions.js]: Am inlocuit cu getIdIndex

**Comment by ancaolteanu** [comment]
Mai trebuie refacute testele pentru noile refactorizari.

**Review by h-mihail — APPROVED** [review]

**Review by msal96 — APPROVED** [review]

**Review by climandrei94 — APPROVED** [review]

**Pull request merged**